src/index.js

```diff
@@ -90,7 +93,7 @@ class DelegatedContentRouting {
     const keyString = key.toBaseEncodedString()
     log('provide starts: ' + keyString)
     await this._httpQueueRefs.add(() =>
-      this.refs(keyString, { recursive: false })
+      all(this.refs(keyString, { recursive: false }))
```

Comment:
We need to return this call, it wasn't being done before either. This is going to add it to the queue, but it will immediately resolve because it's not awaiting or returning the promise.

Reply:
You are right! Added the return now